website/src/components/FlaggableElement.tsx

```diff
@@ -0,0 +1,205 @@
+  };
+
+  return (
+    <Popover
+      isOpen={isEditing}
+      onOpen={setIsEditing.on}
+      onClose={setIsEditing.off}
+      closeOnBlur={false}
+      isLazy
+      lazyBehavior="keepMounted"
+    >
+      <div className="inline-block float-left">
+        <PopoverAnchor>{props.children}</PopoverAnchor>
+        <PopoverTrigger>
+          <Button h="20px">
+            <FlagIcon className="h-5 w-5 ml-3 text-gray-400 group-hover:text-gray-500" aria-hidden="true" />
+          </Button>
+        </PopoverTrigger>
+      </div>
+
+      <PopoverContent width="fit-content">
+        <PopoverArrow />
+        <PopoverCloseButton />
+        <div className="flex mt-3 ">
+          <PopoverBody>
+            <ul>
+              {TEXT_LABEL_FLAGS.map((option, i) => {
+                return <FlagCheckboxLi option={option} key={i} state={label_refs[i]}></FlagCheckboxLi>;
+              })}
+            </ul>
+            <div className="flex justify-center ml-auto">
+              <Button
+                onClick={() => submitResponse(fetchData())}
+                className="bg-indigo-600 text-black hover:bg-indigo-700"
+              >
+                Report
+              </Button>
+            </div>
+          </PopoverBody>
+        </div>
+      </PopoverContent>
+    </Popover>
+  );
+};
+function FlagCheckboxLi(props: { option: textFlagLabels; state: FlagRefs }): JSX.Element {
+  let AdditionalExplanation = null;
+  if (props.option.additionalExplanation) {
+    AdditionalExplanation = (
+      <a href="#" className="group flex items-center space-x-2.5 text-sm ">
+        <QuestionMarkCircleIcon
+          className="flex h-5 w-5 ml-3 text-gray-400 group-hover:text-gray-500"
+          aria-hidden="true"
+        />
+      </a>
+    );
+  }
+
+  return (
+    <>
+      <li>
+        <Flex>
+          <Checkbox
+            onChange={(e) => {
+              props.state.setChecked(e.target.checked);
+            }}
+          />
+          <label
+            className=" ml-1 mr-1 text-sm form-check-label  hover:cursor-pointer"
+            htmlFor={props.option.attributeName}
+          >
+            <span className="text-gray-800 hover:text-blue-700 float-left">{props.option.labelText}</span>
+            {AdditionalExplanation}
+          </label>
+          <Spacer />
+          <Slider
+            width="100px"
+            isDisabled={!props.state.isChecked}
+            defaultValue={100}
+            onChangeEnd={(val) => {
+              props.state.setValue(val / 100);
+            }}
+          >
+            <SliderTrack>
+              <SliderFilledTrack />
+              <SliderThumb />
+            </SliderTrack>
+          </Slider>
+        </Flex>
+      </li>
+    </>
+  );
+}
+interface textFlagLabels {
+  attributeName: string;
+  labelText: string;
+  additionalExplanation?: string;
+}
+const TEXT_LABEL_FLAGS: textFlagLabels[] = [
```

Comment:
this is the same as in `rate_summary.tsx` right? should we extract it to some shared variable?

Reply:
I think the simpler thing to do is in a follow up PR delete the code in `rate_summary.tsx` when this component is added to it.